src/main/java/org/springframework/data/neo4j/config/Neo4jEntityScanner.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2011-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.neo4j.config;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apiguardian.api.API;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.data.annotation.Persistent;
+import org.springframework.data.neo4j.core.schema.Node;
+import org.springframework.data.neo4j.core.schema.RelationshipProperties;
+import org.springframework.lang.Nullable;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * A utility class providing a way to discover an initial entity set for a {@link org.springframework.data.neo4j.core.mapping.Neo4jMappingContext}.
+ *
+ * @author Michael J. Simons
+ * @soundtrack Kelis - Tasty
+ * @since 6.0.2
+ */
+@API(status = API.Status.STABLE, since = "6.0.2")
+public final class Neo4jEntityScanner {
```

Comment:
I would leave it public to be used in the Spring Boot autoconfiguration. The type checks at the end would then be in one single place and tested within SDN. This would avoid having a duplicated logic in SDN and Boot.

Reply:
Yep. It was provided before on our config support. The config support is helpful in a non-spring-boot environment, but not ideal for creating multiple contexts. So it is easier to provide the method on it's own. An example of it's usage is seen in the `Domain2Config` of the multiple context it.

It also can be used from boot…